src/inc/quic_datapath.h

```diff
@@ -709,6 +709,26 @@ CxPlatResolveRoute(
     _In_ CXPLAT_ROUTE_RESOLUTION_CALLBACK_HANDLER Callback
     );
 
+//
+// Set a hint flag to indicate that this packet *should* be sent immediately.
+//
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+CxPlatSendDataSetInlineHint(
+    _In_ CXPLAT_SEND_DATA* SendData
+    );
+
+//
+// Save the processor index to be used later when queuing TX to datapath as an
+// optimization.
+//
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+CxPlatSendDataProcIndex(
+    _In_ CXPLAT_SEND_DATA* SendData,
+    _In_ uint16_t ProcIndex
+    );
```

Comment:
1. You're going to have to implement `CxPlatSendDataProcIndex` for the other data paths.
2. The proc index is also being passed in the send call, right? It should be one place or the other. Not both.
3. Instead of wrapping the calls to `CxPlatSendDataSetInlineHint` with `#ifdef`s, just implement it in the other data paths too, with a no-op. The compiler should be smart enough to make it efficient.

Reply:
Or, maybe move the `#ifdef`s here. On non-raw datapaths, define these to be empty macros, or perhaps undefined_parameter calls.